expression/constant_propagation.go

```diff
@@ -190,20 +190,6 @@ func (s *propConstSolver) propagateConstantEQ() {
 		}
 		for i, cond := range s.conditions {
 			if !visited[i] {
-				// Make sure the collations of the argument are equal.
```

Comment:
Don't we need to check the collation here anymore?

Reply:
The check here is not enough, you can see the example in tests, and the check is moved to column substitution.  Then it will apply to outer join too.